### PR TITLE
peer_info: support multiple peer certificate chains

### PR DESCRIPTION
**Pull request by longlongyang**
1. change peer_cert_chain in struct SpdmPeerInfo to hold
[Option\<SpdmCertChain\>; 8] instread of SpdmCertChain

Signed-off-by: Longlong Yang <longlong.yang@intel.com>

### CODE REVIEW

File: spdmlib/src/requester/psk_finish_req.rs [hunk 1]
```diff
@@ -60,7 +60,7 @@ impl<'a> RequesterContext<'a> {
 
         let transcript_data =
             self.common
-                .calc_req_transcript_data(true, message_k, Some(&message_f))?;
+                .calc_req_transcript_data(0, true, message_k, Some(&message_f))?;
```
Comment (jyao1) [inline spdmlib/src/requester/psk_finish_req.rs:63]: why hardcode 0?

File: spdmlib/src/requester/psk_exchange_req.rs [hunk 2]
```diff
@@ -140,7 +140,7 @@ impl<'a> RequesterContext<'a> {
                         // create session - generate the handshake secret (including finished_key)
                         let th1 = self
                             .common
-                            .calc_req_transcript_hash(true, &message_k, None)?;
+                            .calc_req_transcript_hash(0, true, &message_k, None)?;
```
Comment (jyao1) [inline spdmlib/src/requester/psk_exchange_req.rs:143]: why hardcode 0?
I recommend use INVALID_SLOT, to indicate this is PSK.

File: spdmlib/src/requester/psk_finish_req.rs (hunk 1 above)
Comment (longlongyang) [reply spdmlib/src/requester/psk_finish_req.rs:63]: > 

Good idea!

File: spdmlib/src/requester/psk_exchange_req.rs (hunk 2 above)
Comment (longlongyang) [reply spdmlib/src/requester/psk_exchange_req.rs]: > why hardcode 0? I recommend use INVALID_SLOT, to indicate this is PSK.

Updated! please check the latest commit.

File: spdmlib/src/common/mod.rs [hunk 3]
```diff
@@ -34,6 +34,9 @@ use core::convert::TryInto;
 /// https://www.dmtf.org/sites/default/files/standards/documents/DSP0274_1.1.0.pdf
 pub const ST1: usize = 1_000_000;
 
+/// used as parameter to be slot_id when use_psk is true
+pub const INVALID_SLOT: u8 = 0;
```
Comment (jyao1) [inline spdmlib/src/common/mod.rs:38]: can you define INVALID_SLOT to be 0xFF?

File: spdmlib/src/common/mod.rs [hunk 4]
```diff
@@ -182,9 +186,22 @@ impl<'a> SpdmContext<'a> {
             .append_message(self.runtime_info.message_a.as_ref())
             .ok_or(spdm_err!(ENOMEM))?;
         debug!("message_a - {:02x?}", self.runtime_info.message_a.as_ref());
+
+        if self.peer_info.peer_cert_chain[slot_id as usize].is_none() && !use_psk {
```
Comment (jyao1) [inline spdmlib/src/common/mod.rs:190]: I think this check should be moved into !use_psk

File: spdmlib/src/common/mod.rs (hunk 3 above)
Comment (longlongyang) [reply spdmlib/src/common/mod.rs]: > can you define INVALID_SLOT to be 0xFF?

Updated.

File: spdmlib/src/common/mod.rs (hunk 4 above)
Comment (longlongyang) [reply spdmlib/src/common/mod.rs]: > I think this check should be moved into !use_psk

Updated.

**Pull request merged**